crowbar_framework/app/models/cinder_service.rb

```diff
@@ -205,6 +205,26 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
     # the VIP of the cluster to be setup
     allocate_virtual_ips_for_any_cluster_in_networks(controller_elements, vip_networks)
 
+    # Generate secrets uuid for libvirt rbd backend
+    dirty = false
+    proposal = ProposalObject.find_data_bag_item "crowbar/bc-cinder-default"
+    role.default_attributes[:cinder][:volumes].each_with_index do |volume, volid|
+    # proposal[:attributes][:cinder][:volumes].each_with_index do |volume, volid|
+      next unless volume[:backend_driver] == "rbd"
+      if volume[:rbd][:secret_uuid].empty?
+        secret_uuid = `uuidgen`.strip
+        volume[:rbd][:secret_uuid] = secret_uuid
+        proposal[:attributes][:cinder][:volumes][volid][:rbd][:secret_uuid] = secret_uuid
+        dirty = true
+      end
+    end
+    # Save only if there was change
```

Comment:
I wouldn't keep that comment line as it's kind of obvious, but you can keep it if you prefer :-)